### DISCUSSION TITLE
How to make camera ignore entity?

### DISCUSSION THREAD

**Discussion by naomijub** [bevyengine/bevy#11114] [Q&A]
So, I've been working on this [PR](https://github.com/rewin123/space_editor/pull/135), which has at least 3 cameras, play mode camera, editor camera and simulate playmode camera in another view tab. The issue is that I want to ignore debug_grid in the newly created camera, but not in the editor camera.

I remember that around version 0.8 I could do it with render pass, but I'm not sure how to do it now

**Comment by torsteingrindvik** [comment]
Perhaps you can solve it with render layers: https://github.com/bevyengine/bevy/blob/8067e46049f222d37ac394745805bad98979980f/examples/3d/render_to_texture.rs#L75

If the debug grid is on layer _n_, you can opt-in to rendering this layer on a per-camera basis by adding layer _n_ to those cameras as well afaik.

The example shows that the last camera pass uses the default layer 0, so it does not see the layer 1 stuff earlier in the example.

**Reply by naomijub** [reply]
Is there a way to just opt out a layer? I feel that defining layers this way could become very rigid for other users

**Reply by torsteingrindvik** [reply]
Not sure I'm following what you mean?

If I understood your problem, you could:

1. Add the debug grid to render layer e.g. 1
2. Make editor camera render this layer
3. Don't add this layer to the new camera, it won't render the debug grid then since it's not on the default 0 layer

**Reply by naomijub** [reply]
Hey, tried doing this but I seemed to have done something wrong, can you check the PR and help?
The commit hash is 239e474 and this is the image I have right now when adding a playmode camera and a mesh obj.
<img width="1191" alt="Captura de Tela 2023-12-28 às 6 25 09 PM" src="https://github.com/bevyengine/bevy/assets/14813660/7584938d-de40-4729-bfa2-b82e37702343">


https://github.com/rewin123/space_editor/pull/135

**Comment by torsteingrindvik** [comment]
Hm I can't easily tell by just looking at the commit.

Gut feeling is that perhaps the combination of

- Render layers
- Not clearing cameras (I saw clear color none)
- Render ordering

might have hit a but somewhere.
